smithy-typescript-codegen/src/main/java/software/amazon/smithy/typescript/codegen/WaiterGenerator.java

```diff
@@ -72,7 +72,9 @@ private void generateWaiter() {
         writer.addImport("WaiterState", "WaiterState", WAITABLE_UTIL_PACKAGE);
         writer.addImport("WaiterConfiguration", "WaiterConfiguration", WAITABLE_UTIL_PACKAGE);
 
+        // generates the now deprecated WaitFor....
         writer.writeDocs(waiter.getDocumentation().orElse("") + " \n"
+                + " @deprecated in favor of waitUntil" + waiterName + ". This does not throw on failure.\n"
                 + " @param params : Waiter configuration options.\n"
                 + " @param input : the input to " + operationSymbol.getName() + " for polling.");
```

Comment:
Can you add the lines of `@params` in to the `getDocumentation()`? `@deprecated` can be a separateline. This can make code simpler. Please also make sure the doc is generated correctly that the function is deprecated instead of paramter.

Reply:
Not sure what you mean here

Reply:
The 2 line with `@param` are generated duplicatedly here and bellow. I was meant to generate them in `watier.getDocumentation()`. But Now I realize that they are from Smithy model. So this comment is no longer applicable.